src/main/scala/mesosphere/marathon/upgrade/ReadinessBehavior.scala

```diff
@@ -35,16 +35,35 @@ trait ReadinessBehavior { this: Actor with ActorLogging =>
   val versionString: String = version.toString
 
   //state managed by this behavior
-  var healthy = Set.empty[Task.Id]
-  var ready = Set.empty[Task.Id]
-  var subscriptions = Map.empty[String, Subscription]
+  private[this] var healthy = Set.empty[Task.Id]
+  private[this] var ready = Set.empty[Task.Id]
+  private[this] var subscriptions = Map.empty[ReadinessCheckSubscriptionKey, Subscription]
 
   /**
     * Hook method which is called, whenever a task becomes ready according to the given app definition.
+    *
     * @param taskId the id of the task that has become ready.
     */
   def taskIsReady(taskId: Task.Id): Unit
 
+  /**
+    * Call this method, if a task terminates.
```

Comment:
"Actors extending this trait should call this method when they detect that a task is terminated"

I think we also need an empty line after the "brief description" and maybe also between the details and the params.